internal/p2p/router.go

```diff
@@ -688,11 +687,17 @@ func (r *Router) connectPeer(ctx context.Context, address NodeAddress) {
 		return
 	}
 
-	_, _, err = r.handshakePeer(ctx, conn, address.NodeID)
+	peerInfo, _, err := r.handshakePeer(ctx, conn, address.NodeID)
+	var errRejected ErrRejected
 	switch {
 	case errors.Is(err, context.Canceled):
 		conn.Close()
 		return
+	case errors.As(err, &errRejected) && errRejected.IsIncompatible():
+		r.logger.Error("peer rejected due to incompatibility", "node", peerInfo.NodeID, "err", err)
+		r.peerManager.Errored(peerInfo.NodeID, err)
```

Comment:
Ok so turns out you don't really need this because the peer manager only accepts the peer afterwards so according to the peer manager we weren't even connected in the first place.

Reply:
Second thing is that this is just for outbound connections. We should also check for compatibility in the inbound connections (see `openConnection`). I see you've already added a test case for inbound peers so I'm a little confused where the check happens but I will look into it.

Reply:
I'm still unsure about reusing the `ErrRejected` error since in the next release we will mostly remove it along with the `Switch` so I don't know if this will warrant its own error but we can leave it for now and come back to it later.

Reply:
Ah for the inbound connections, I see that we don't need to add this specific case because the next case in the switch statement is `err != nil` which catches it anyway

Reply:
I think we can probably remove this case then as it is pretty redundant